test/csuite/timestamp_abort/main.c

```diff
@@ -478,21 +482,21 @@ backup_create_incremental(WT_CONNECTION *conn, uint32_t src_index, uint32_t inde
     nranges = 0;
     nunmodified = 0;
 
-    testutil_check(
-      __wt_snprintf(src_backup_home, sizeof(src_backup_home), BACKUP_BASE "%" PRIu32, src_index));
+    testutil_snprintf(src_backup_home, sizeof(src_backup_home), BACKUP_BASE "%" PRIu32, src_index);
 
     printf("Create incremental backup %" PRIu32 " - start: source=%" PRIu32 "\n", index, src_index);
 
     /* Prepare the directory. */
-    testutil_check(__wt_snprintf(backup_home, sizeof(backup_home), BACKUP_BASE "%" PRIu32, index));
-    testutil_system("rm -rf %s && mkdir %s", backup_home, backup_home);
+    testutil_snprintf(backup_home, sizeof(backup_home), BACKUP_BASE "%" PRIu32, index);
+    testutil_remove(backup_home);
+    testutil_mkdir(backup_home);
```

Comment:
I am in favor of this. There are a few other programs that also do a similar action so they should be updated too as part of this. There are A LOT of programs that use `wt_snprintf` so fixing that could be a separate ticket. I would prefer the `remove` and `mkdir` be done together here I think.

Reply:
>  I am in favor of this.

Sounds good! I'll add appropriate calls to `testutil_snprintf` as I go ahead and replace other `system` calls by the new functions, and then create a ticket for the rest.

>  I would prefer the `remove` and `mkdir` be done together here I think.

Just to clarify, are you suggesting that we will create a new function called something like `testutil_recreate_dir`, which would perform `remove` and then `mkdir`? Or should we replace all calls to functions like `testutil_make_work_dir` by calls to these two functions?

I feel that this is such a common pattern, that it wouldn't hurt to have a single function that does both `remove` and `mkdir`, and then we can use it instead of `testutil_make_work_dir` as you suggested above.

Reply:
I made three different suggestions:
1. New ticket for `wt_snprintf` to `testutil_snprintf`.
2. Get rid of `testutil_make|clean_work_dir` since they're just one line and use `testutil_remove` (maybe call it `testutil_rmdir`??) and `testutil_mkdir`. Fix all the csuite tests that use them.
3. Go through other `test/csuite` programs and fix the `wt_snprintf(..."rm -rf %s && mkdir %s"); system(buf)` type calls and replace them like you did in `timestamp_abort`.

Reply:
Thanks for clarifying!

The current implementation of `testutil_make_work_dir` is actually two lines, which is why I was suggesting to rename it to something like `testutil_recreate_dir` instead of just removing it, as rename followed by create is a common pattern:
```
void
testutil_make_work_dir(const char *dir)
{
    testutil_clean_work_dir(dir);
    testutil_mkdir(dir);
}
```

The reasons why I named the remove function as `testutil_remove` instead of `rmdir` is because it can also remove files (potentially even multiple files, as it expands the glob patterns), and because the commad-line utility `rmdir` only handles empty directories, so I didn't want to cause any confusion. But it is true that all current (and likely future) uses of remove are for directories, so I can see that side of the argument too.

Reply:
Ah - I missed the `testutil_clean_work_dir` line. With that, I like your suggestion of naming it `testutil_recreate_dir` and getting rid of `testutil_clean_work_dir` and just using `testutil_remove`. I am fine with the name `remove` instead of `rmdir` given your explanation.